dds_web/api/project.py

```diff
@@ -229,37 +229,29 @@ class GetPrivate(flask_restful.Resource):
     @logging_bind_request
     def get(self):
         """Get private key from database"""
-
+        # Verify access to project
         project = project_schemas.ProjectRequiredSchema().load(flask.request.args)
 
-        # TODO (ina): Change handling of private key -- not secure
         flask.current_app.logger.debug("Getting the private key.")
 
-        app_secret = flask.current_app.config.get("SECRET_KEY")
-        passphrase = app_secret.encode("utf-8")
-
-        enc_key = bytes.fromhex(project.private_key)
-        nonce = bytes.fromhex(project.privkey_nonce)
-        salt = bytes.fromhex(project.privkey_salt)
+        # Get Project keys row for current user and project
+        project_keys_row = models.ProjectKeys.query.filter_by(
+            project_id=project.id, user_id=auth.current_user().username
+        ).one_or_none()
+        if not project_keys_row:
+            raise AccessDeniedError("Project key missing.")
```

Comment:
If-checks make the code a bit crowded. There is a KeyNotFoundError implemented in obtain_project_private_key. We could perhaps rewrite that better, if you like. So I think lines 241-242 could be removed.

Reply:
@worukan I like the exception you added too but I'd like to keep this if statement for now at least. It's more explicit. It's possible we will remove this though but for now we can keep it.